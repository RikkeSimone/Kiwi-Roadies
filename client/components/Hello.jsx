import React from 'react'
import { connect } from 'react-redux'
import { acquireName } from '../actions'

let name = ''

class Hello extends React.Component {
  render () {
    return (
      <>
        { this.props.name
          ? null
          : <div className="hello-container">
            <div>
              <div className="intro">
                <img className="intro-message" src="./up-for-a-roadie.png"/>
              </div>
              <h3>WHAT&apos;S YOUR NAME?</h3>
              <div className="form-container">
                <div className="form__group">
                  <input type="text" className="form__input" id="name" placeholder="Your name" required=""
                    onChange={(event) => (name = event.target.value)}
                  />
                </div>
                <span className="fas fa-arrow-circle-right" onClick={() => this.props.dispatch(acquireName(name))}></span>
              </div>
            </div>
          </div>
        }
      </>
    )
  }
}

const mapStateToProps = state => {
  return {
    name: state.name
  }
}

export default connect(mapStateToProps)(Hello)
