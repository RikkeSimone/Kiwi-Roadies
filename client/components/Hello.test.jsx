import React from 'react'
import { renderWithRedux } from '../testing/utils'
import '@testing-library/jest-dom'
import 'regenerator-runtime/runtime'

import Hello from './Hello'

test('Hello renders the expected items', async () => {
  const { asFragment } = renderWithRedux(<Hello />)
  expect(asFragment()).toMatchSnapshot()
})
