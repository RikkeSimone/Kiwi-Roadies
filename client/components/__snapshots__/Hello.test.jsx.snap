// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello renders the expected items 1`] = `
<DocumentFragment>
  <div
    class="hello-container"
  >
    <div>
      <div
        class="intro"
      >
        <p
          class="intro-message"
        >
          UP FOR A 
          <br />
          ROADIE?
        </p>
      </div>
      <div
        class="name-q"
      >
        <h3>
          WHAT'S YOUR NAME?
        </h3>
      </div>
      <div
        class="form-container"
      >
        <div
          class="form__group"
        >
          <input
            class="form__input"
            id="name"
            placeholder="Your name"
            type="text"
          />
        </div>
        <span
          class="fas fa-arrow-circle-right"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
